ts/src/program/common.ts

```diff
@@ -58,8 +58,10 @@ export function translateAddress(address: Address): PublicKey {
   if (typeof address === "string") {
     const pk = new PublicKey(address);
     return pk;
-  } else {
+  } else if (address.constructor.prototype.constructor.name === "PublicKey") {
```

Comment:
This has introduced some bugs in our production build. I believe this is because JS is minifying the code and changing the name of constructors. 

A couple suggestions to avoid this:

- check if the method `toBase58` exists on the object
- add a `name` or `type` to the `PublicKey` constructor object similar to how Phantom adds `isPhantom` to `window.solana`

What do you think? Happy to open a PR that fixes this :)

cc @paul-schaaf

Reply:
sorry for that, didnt consider this, my mistake!

fix is already on the way https://github.com/project-serum/anchor/pull/1138

Reply:
Awesome, thanks for the pointer there @paul-schaaf :)

Reply:
I don't have the full picture of the codebase and the issue, just wanted to chime in that some time ago I fixed precisely the same bug replacing such a clause with an `instanceof` — `address instanceof PublicKey`.

Reply:
we can try again but it didnt work for us before https://github.com/project-serum/anchor/issues/1101